src/daemon.rs

```diff
@@ -48,11 +48,12 @@ fn rpc_poll(client: &mut bitcoincore_rpc::Client) -> PollResult {
 
 pub(crate) fn rpc_connect(config: &Config) -> Result<bitcoincore_rpc::Client> {
     let rpc_url = format!("http://{}", config.daemon_rpc_addr);
-    if !config.daemon_cookie_file.exists() {
-        bail!("{:?} is missing", config.daemon_cookie_file);
+    if let bitcoincore_rpc::Auth::CookieFile(ref path) = config.daemon_auth {
+        if !path.exists() {
+            bail!("{:?} is missing", path);
+        }
```

Comment:
Isn't it better to let RPC code check this? This is TOCTOU issue and I don't see any additional benefit.

Reply:
Unfortunately, RPC error message is not very clear in case the cookie file is missing:
```
Error: failed to connect to RPC: 127.0.0.1:8332

Caused by:
    I/O error: No such file or directory (os error 2)
```

Reply:
Yeah, io::Error not having path is the single most annoying thing about Rust. :D
Let's keep it this way and fix the error in a separate issue/PR.